[ADD] add a new module to allow planner changing the supply method on procurement order level

duanyp1991 opened the pull request:
[ADD] add a new module to allow planner changing the supply method on procurement order level
 Changes to be committed:

```
new file:   procurement_supply_method/__init__.py
new file:   procurement_supply_method/__openerp__.py
new file:   procurement_supply_method/procurement.py
new file:   procurement_supply_method/procurement_supply_method_views.xml
```

Commit:
[ADD] add a new module to allow planner changing the supply method on the procurement order level.
 Changes to be committed:
	new file:   procurement_supply_method/__init__.py
	new file:   procurement_supply_method/__openerp__.py
	new file:   procurement_supply_method/procurement.py
	new file:   procurement_supply_method/procurement_supply_method_views.xml

clonedagain commented on procurement_supply_method/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+    'website': 'www.openerp.net.cn',
+    'category': 'Generic Modules/Production',
+    'depends': ['procurement', 'sale', 'mrp', 'purchase'],
+    'description': """
```
Can you please use the OCA template? https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

duanyp1991 replied:
@clonedagain Hi, The template you provided is in the syntax of *rst, but the structure is quite good, I'll make another commit to change the description of the module, thanks a lot!

clonedagain replied:
As you are targeting v7, you can use RST in the description and it should be rendered correctly.
Later when you migrate the module to v8, remember to replace the description in this file by a separate `README.rst` file.

duanyp1991 replied:
@clonedagain Okay, thanks for the tip! :)

clonedagain commented on procurement_supply_method/procurement.py:
```diff
@@ -0,0 +1,138 @@
+                    cr, uid, ids, context=context)
+
+
+class SaleOrder(osv.osv):
```
Would you care to move this to a `sale.py`?

clonedagain commented on procurement_supply_method/procurement.py:
```diff
@@ -0,0 +1,138 @@
+        return True
+
+
+class MrpProduction(osv.osv):
```
... and move this to `mrp.py`?

clonedagain commented:
Does that mean the person in charge must review/confirm each procurement before they can run?

Would you care to add a concrete use case for this module in the description?

gurneyalex commented on procurement_supply_method/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+{
+    'name': 'Procurement Improvement - supply method',
+    'version': '1.0',
+    'author': 'Elico Corp',
```
please add "Odoo Community Association (OCA)"

Commit:
[IMP]
    * improvement on description in file: __openerp__.py
        - using communinty template
        - adding business case
    * add maintainer:OCA
 Changes to be committed:
	modified:   procurement_supply_method/__init__.py
	modified:   procurement_supply_method/__openerp__.py
	new file:   procurement_supply_method/mrp.py
	modified:   procurement_supply_method/procurement.py
	new file:   procurement_supply_method/sale.py

duanyp1991 commented:
@gurneyalex I have added the OCA maintainer.:) 
@clonedagain I have improved the description and added one business case. About the flask8 checking, the number of characters of one line should be less than 100 instead of 79, no?
Thanks.

gurneyalex commented:
@duanyp1991 we are using 79 for all addons, after a long discussion.  Unless PEP8 has been updated, we are sticking to this. 

Thanks for updating the author.

gurneyalex commented on procurement_supply_method/__openerp__.py:
```diff
@@ -0,0 +1,109 @@
+        on the procurement order level instead of having two products with different
+        supply methods.
+
+Please note that this module re-define the field: procure_method
```
s/define/defines/

duanyp1991 replied:
@gurneyalex  Thanks! already corrected.

gurneyalex commented on procurement_supply_method/__openerp__.py:
```diff
@@ -0,0 +1,109 @@
+    'website': 'www.openerp.net.cn',
+    'category': 'Generic Modules/Production',
+    'depends': ['procurement', 'sale', 'mrp', 'purchase'],
+    'description': """
```
I suggest removing the key from the metadata dictionary, and moving the description to a file called README.rst, as it will make the reading easier.

duanyp1991 replied:
@gurneyalex thanks for your note! I have added the file README.rst according to your advise, I still keep the description in the metadata dict since people can still check the module information through the Odoo interface.

gurneyalex commented on procurement_supply_method/procurement.py:
```diff
@@ -0,0 +1,102 @@
+        # make this field editable when in state draft, confirmed and exception.
+        'procure_method': fields.selection(
+            [('make_to_stock', _('Make to Stock')), ('make_to_order', _('Make to Order'))],
+            _('Procurement Method'),
```
I think the calls to _() are implicit here, so they can be removed.

duanyp1991 replied:
@gurneyalex  well noted. Thanks.

Commit:
[MOD]:
	* Fix pep8
	* Add README.rst file
	* change typo

pedrobaeza commented:
@duanyp1991 are you still interested in this PR?

LoisRForgeFlow commented:
No answer. Closing, please feel free to reopen.
